x-pack/plugins/cases/server/saved_object_types/migrations/comments.ts

```diff
@@ -119,39 +119,54 @@ export const createCommentsMigrations = (
   return mergeSavedObjectMigrationMaps(commentsMigrations, embeddableMigrations);
 };
 
-export const migrateByValueLensVisualizations =
-  (migrate: MigrateFunction): SavedObjectMigrationFn<{ comment?: string }, { comment?: string }> =>
-  (doc: SavedObjectUnsanitizedDoc<{ comment?: string }>, context: SavedObjectMigrationContext) => {
-    if (doc.attributes.comment == null) {
-      return doc;
-    }
+export const migrateByValueLensVisualizations = (
+  migrate: MigrateFunction,
+  migrationVersion: string
+): SavedObjectMigrationParams<{ comment?: string }, { comment?: string }> => {
+  const deferred = isDeferredMigration(MIN_DEFERRED_KIBANA_VERSION, migrationVersion);
 
-    try {
-      const parsedComment = parseCommentString(doc.attributes.comment);
-      const migratedComment = parsedComment.children.map((comment) => {
-        if (isLensMarkdownNode(comment)) {
-          // casting here because ts complains that comment isn't serializable because LensMarkdownNode
-          // extends Node which has fields that conflict with SerializableRecord even though it is serializable
-          return migrate(comment as SerializableRecord) as LensMarkdownNode;
-        }
-
-        return comment;
-      });
-
-      const migratedMarkdown = { ...parsedComment, children: migratedComment };
+  return {
```

Comment:
We now change "normal" lens migrations into deferred ones. This would work, but I think it'd be more explicit if the actual lens migration was also forced to set `deferred: true`.

To do this we'd need to handle migrate being `migrate: MigrateFunction | SavedObjectMigrationParams`

And then do something like:
```
if(deferred) { // nit: I would call this variable shouldDefer 
  if(typeof(migrate) == "function" || migrate.deferred === false) { // just for illustration, typescript won't like this
   throw new Error("Cases only allows deferred migrations starting in 8.9.0")
  }
}
```

Reply:
Discussed this with @cnasikas and @stratoula 

This would require a small refactor in embeddable migration framework to allow for `MigrateFunction | SavedObjectMigrationParams` instead of just `MigrateFunction`. While we ultimately want all embeddable migrations to be deferred we're likely going to start moving to a client-side embeddable migration framework soon https://github.com/elastic/kibana/issues/158677

So for now we decided we won't force the lens migration to be `deferred: true` but only apply that when we migrate the case comments.